test/lib/assert-equal-dom.js

```diff
@@ -31,7 +32,12 @@ function areEqual(a, b) {
                 return equalStyle(a[key], b[key])
             }
             if (typeof a === "object" || typeof a === "function") {
-                if (!areEqual(a[key], b[key])) {
+                if (cache.indexOf(a) !== -1) {
```

Comment:
I want to support IE8 at least because I know people out there in the real world still have to suffer through it. It's not like we are sacrificing features.

Basically, there is some property here that needs to be excluded from assertEqualDom. This should be fixed by finding which key contains the circular reference and ignoring it (shouldn't be difficult to find, just add some log statements and see what gets printed infinitely).